# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-lt-build flac-build gdk-pixbuf-build gst-plugins-base-build speex-build"

PKG_CONFIG_PATH=/fb/share/pkgconfig pkg-config --exists --print-errors $REQUIRES || exit 1

# configure (ac-2.69; am-1.15; lt-2.4.2)
libtoolize -c -f; aclocal -I m4 -I common/m4; autoconf; autoheader -f; automake -a -c
PKG_CONFIG_PATH=/fb/lib/pkgconfig LDFLAGS=-Wl,-rpath,/fb/lib ./configure --prefix=/fb

# build
make

# install
install -d $DESTDIR/fb/lib/gstreamer-1.0
install ext/cairo/.libs/libgstcairo.so $DESTDIR/fb/lib/gstreamer-1.0
install ext/flac/.libs/libgstflac.so $DESTDIR/fb/lib/gstreamer-1.0
install ext/gdk_pixbuf/.libs/libgstgdkpixbuf.so $DESTDIR/fb/lib/gstreamer-1.0
install ext/jpeg/.libs/libgstjpeg.so $DESTDIR/fb/lib/gstreamer-1.0
install ext/libpng/.libs/libgstpng.so $DESTDIR/fb/lib/gstreamer-1.0
install ext/speex/.libs/libgstspeex.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/law/.libs/libgstalaw.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/alpha/.libs/libgstalpha.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/alpha/.libs/libgstalphacolor.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/apetag/.libs/libgstapetag.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/audiofx/.libs/libgstaudiofx.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/audioparsers/.libs/libgstaudioparsers.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/auparse/.libs/libgstauparse.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/autodetect/.libs/libgstautodetect.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/avi/.libs/libgstavi.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/cutter/.libs/libgstcutter.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/debugutils/.libs/libgstdebug.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/deinterlace/.libs/libgstdeinterlace.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/dtmf/.libs/libgstdtmf.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/effectv/.libs/libgsteffectv.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/equalizer/.libs/libgstequalizer.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/flv/.libs/libgstflv.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/flx/.libs/libgstflxdec.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/goom/.libs/libgstgoom.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/goom2k1/.libs/libgstgoom2k1.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/icydemux/.libs/libgsticydemux.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/id3demux/.libs/libgstid3demux.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/imagefreeze/.libs/libgstimagefreeze.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/interleave/.libs/libgstinterleave.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/isomp4/.libs/libgstisomp4.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/level/.libs/libgstlevel.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/matroska/.libs/libgstmatroska.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/law/.libs/libgstmulaw.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/multifile/.libs/libgstmultifile.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/multipart/.libs/libgstmultipart.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/debugutils/.libs/libgstnavigationtest.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/replaygain/.libs/libgstreplaygain.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/rtp/.libs/libgstrtp.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/rtpmanager/.libs/libgstrtpmanager.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/rtsp/.libs/libgstrtsp.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/shapewipe/.libs/libgstshapewipe.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/smpte/.libs/libgstsmpte.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/spectrum/.libs/libgstspectrum.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/udp/.libs/libgstudp.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/videobox/.libs/libgstvideobox.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/videocrop/.libs/libgstvideocrop.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/videofilter/.libs/libgstvideofilter.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/videomixer/.libs/libgstvideomixer.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/wavenc/.libs/libgstwavenc.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/wavparse/.libs/libgstwavparse.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/y4m/.libs/libgsty4menc.so $DESTDIR/fb/lib/gstreamer-1.0
install sys/oss/.libs/libgstossaudio.so $DESTDIR/fb/lib/gstreamer-1.0
install sys/oss4/.libs/libgstoss4audio.so $DESTDIR/fb/lib/gstreamer-1.0
install sys/v4l2/.libs/libgstvideo4linux2.so $DESTDIR/fb/lib/gstreamer-1.0

# build.pc
install -d $DESTDIR/fb/share/pkgconfig
cat > $DESTDIR/fb/share/pkgconfig/gst-plugins-good-build.pc << EOF
Name: gst-plugins-good
Version: 1.6.4
Description: Good GStreamer plugins
Requires: $REQUIRES

exec=\\
/fb/lib/gstreamer-1.0/libgstalaw.so \\
/fb/lib/gstreamer-1.0/libgstalpha.so \\
/fb/lib/gstreamer-1.0/libgstalphacolor.so \\
/fb/lib/gstreamer-1.0/libgstapetag.so \\
/fb/lib/gstreamer-1.0/libgstaudiofx.so \\
/fb/lib/gstreamer-1.0/libgstaudioparsers.so \\
/fb/lib/gstreamer-1.0/libgstauparse.so \\
/fb/lib/gstreamer-1.0/libgstautodetect.so \\
/fb/lib/gstreamer-1.0/libgstavi.so \\
/fb/lib/gstreamer-1.0/libgstcairo.so \\
/fb/lib/gstreamer-1.0/libgstcutter.so \\
/fb/lib/gstreamer-1.0/libgstdebug.so \\
/fb/lib/gstreamer-1.0/libgstdeinterlace.so \\
/fb/lib/gstreamer-1.0/libgstdtmf.so \\
/fb/lib/gstreamer-1.0/libgsteffectv.so \\
/fb/lib/gstreamer-1.0/libgstequalizer.so \\
/fb/lib/gstreamer-1.0/libgstflac.so \\
/fb/lib/gstreamer-1.0/libgstflv.so \\
/fb/lib/gstreamer-1.0/libgstflxdec.so \\
/fb/lib/gstreamer-1.0/libgstgdkpixbuf.so \\
/fb/lib/gstreamer-1.0/libgstgoom.so \\
/fb/lib/gstreamer-1.0/libgstgoom2k1.so \\
/fb/lib/gstreamer-1.0/libgsticydemux.so \\
/fb/lib/gstreamer-1.0/libgstid3demux.so \\
/fb/lib/gstreamer-1.0/libgstimagefreeze.so \\
/fb/lib/gstreamer-1.0/libgstinterleave.so \\
/fb/lib/gstreamer-1.0/libgstisomp4.so \\
/fb/lib/gstreamer-1.0/libgstjpeg.so \\
/fb/lib/gstreamer-1.0/libgstlevel.so \\
/fb/lib/gstreamer-1.0/libgstmatroska.so \\
/fb/lib/gstreamer-1.0/libgstmulaw.so \\
/fb/lib/gstreamer-1.0/libgstmultifile.so \\
/fb/lib/gstreamer-1.0/libgstmultipart.so \\
/fb/lib/gstreamer-1.0/libgstnavigationtest.so \\
/fb/lib/gstreamer-1.0/libgstossaudio.so \\
/fb/lib/gstreamer-1.0/libgstoss4audio.so \\
/fb/lib/gstreamer-1.0/libgstpng.so \\
/fb/lib/gstreamer-1.0/libgstreplaygain.so \\
/fb/lib/gstreamer-1.0/libgstrtp.so \\
/fb/lib/gstreamer-1.0/libgstrtpmanager.so \\
/fb/lib/gstreamer-1.0/libgstrtsp.so \\
/fb/lib/gstreamer-1.0/libgstshapewipe.so \\
/fb/lib/gstreamer-1.0/libgstsmpte.so \\
/fb/lib/gstreamer-1.0/libgstspectrum.so \\
/fb/lib/gstreamer-1.0/libgstspeex.so \\
/fb/lib/gstreamer-1.0/libgstudp.so \\
/fb/lib/gstreamer-1.0/libgstvideo4linux2.so \\
/fb/lib/gstreamer-1.0/libgstvideobox.so \\
/fb/lib/gstreamer-1.0/libgstvideocrop.so \\
/fb/lib/gstreamer-1.0/libgstvideofilter.so \\
/fb/lib/gstreamer-1.0/libgstvideomixer.so \\
/fb/lib/gstreamer-1.0/libgstwavenc.so \\
/fb/lib/gstreamer-1.0/libgstwavparse.so \\
/fb/lib/gstreamer-1.0/libgsty4menc.so
EOF

exit
--- gst-plugins-good-1.6.4.orig/gst/matroska/matroska-demux.c
+++ gst-plugins-good-1.6.4/gst/matroska/matroska-demux.c
@@ -55,9 +55,7 @@
 
 /* For AVI compatibility mode
    and for fourcc stuff */
-#include <gst/riff/riff-read.h>
-#include <gst/riff/riff-ids.h>
-#include <gst/riff/riff-media.h>
+#include <gst/riff/riff.h>
 
 #include <gst/audio/audio.h>
 #include <gst/tag/tag.h>
--- gst-plugins-good-1.6.4.orig/gst/rtsp/gstrtspsrc.c
+++ gst-plugins-good-1.6.4/gst/rtsp/gstrtspsrc.c
@@ -87,9 +87,8 @@
 #include <stdio.h>
 #include <stdarg.h>
 
-#include <gst/net/gstnet.h>
-#include <gst/sdp/gstsdpmessage.h>
-#include <gst/sdp/gstmikey.h>
+#include <gst/net/net.h>
+#include <gst/sdp/sdp.h>
 #include <gst/rtp/rtp.h>
 
 #include "gst/gst-i18n-plugin.h"
